### PR TITLE
Custom pid errors

### PR DESCRIPTION
**Pull request by cowsed**
Add custom error functions to pid class. 
Error calculation functions are of the form  `double calculate_error(double sensor_val, double target)`. 
Useful in situations where the error is not strictly `target - sensor_val` (angles).
If no custom function is specified, `target - sensor_val` is still used.

### CODE REVIEW

File: core/include/utils/motion_controller.h [hunk 1]
```diff
@@ -43,6 +43,17 @@ class MotionController : public Feedback
      * @param ff_cfg Definitions of kS, kV, and kA
      */
     MotionController(m_profile_cfg_t &config);
+    /**
+     * @brief Construct a new Motion Controller object
+     * 
+     * @param max_v Maximum velocity the movement is capable of
```
Comment (dec1701) [inline core/include/utils/motion_controller.h]: can we make it clear that these parameters are passed through the m_profile_cfg_t struct?

Comment (cowsed) [reply core/include/utils/motion_controller.h]: yes

**Comment by superrm11** [comment]
If this is an issue with strictly angles, it might be better to simplify with an enum option, like:
`PID pid(cfg, Type::LINEAR); //or Type::ANGULAR`
and let the PID or Motion Profile class handle it. Might be less confusing than a function pointer, and invite less error.  Thoughts?

**Comment by cowsed** [comment]
Yeah - and if a need arises for a different custom strategy thats just another enum value.

**Pull request merged**